client/lib/analytics/README.md

```diff
@@ -122,3 +139,16 @@ If we had instead used `calypso_add_cart_product` and `calypso_remove_cart_produ
 Finally, for consistency, the verb at the end should be in the form `add`, `remove`, `view`, `click`, etc and _not_ `adds`, `added`, `adding`, etc.
 
 With the exception of separating tokens with underscores, these rules do not apply to property names. `coupon_code` is perfectly fine.
+
+## Page Views Paths Conventions
+
+There are currently no enforced rules for page views paths reporting but, where possible, they should be normalized and the variables values replaced by placeholders.
+
+E.g. the path `/comments/all/example.wordpress.com/1234` should be reported as `/comments/all/:site/:post_id`.
+
+As a rule of thumb:
+
+- The **site fragment** should be reported simply as `:site`.
+- **IDs** should be reported as `:element_id` (notice the snake case).
+- **Non-enumerable variables** (e.g. the domain name in `/domains/manage/:domain/edit/:site`) should be reported without unnecessary qualifiers.
+- The value of **enumerable variables** should be reported instead of the placeholder where it make sense (e.g. if changing the variable value results in a page change, as it happens for the status in `/comments/spam/:site`).
```

Comment:
While not directly related to page view tracking, I noticed that a lot of page view reporting issues stemmed from very loose matching of routes with `page`, especially when we have a chain of optional path parameters. For example:

`/posts/:author?/:status?/:site?`

Recording page views in this case becomes harder because you have to validate parts of `context.path` and have the logic for building the normalized path that will be passed to `PageViewTracker` in the component. When we have a known set of possible values, I think we should always list them explicitly. Given above example that would look something like this:

`/posts/:author(my|everyone)?/:status(published|drafts|scheduled|trashed)?/:site?`

Now we can safely pass the `context.params.author` and `context.params.status` to the component responsible for rendering `PageViewTracker` and use them in path. In addition, this should almost always be followed by catch-all route (e.g. `/posts/(.*)`) that redirects to root path (e.g. `/posts`), which will also handle recording that page view.

Reply:
Maybe add a how to build new routes readme and we can link to it?

Reply:
Another way to show these are common examples of what to avoid and what to do instead